src/modules/navigator/land.cpp

```diff
@@ -58,7 +58,8 @@ Land::on_activation()
 	/* convert mission item to current setpoint */
 	struct position_setpoint_triplet_s *pos_sp_triplet = _navigator->get_position_setpoint_triplet();
 
-	if (_navigator->get_vstatus()->vehicle_type == vehicle_status_s::VEHICLE_TYPE_ROTARY_WING) {
+	if (_navigator->get_vstatus()->vehicle_type == vehicle_status_s::VEHICLE_TYPE_ROTARY_WING
+	    && _navigator->get_local_position()->xy_global) { // only execute if global position is valid
```

Comment:
This should also be done further down in the file after the VTOL transition then.

Reply:
Done in ea98e1d

Reply:
Yes correct, though a VTOL couldn't fly with local position only in fixed-wing in the first place. Still, doesn't hurt to add this guard.